File: 4kyu/Twice linear/solution.js
function dblLinear(n) {
    var u = [];
    u[0] = 1;
    var a = 0;
    var b = 0;
    while( u.length<n+1 ){
    
      var y = 2 * u[a]+1;
      var z = 3 * u[b]+1;
      if( y<z ){ u.push(y); a++; }
      else if( y>z ){ u.push(z); b++ }
      else{ u.push(z); a++; b++; } 
    } 
    return u[n];
}

File: 4kyu/Twice linear/solution.py
def dbl_linear(n):
    u = [1]
    a = 0
    b = 0
    for i in range(0,n):
        y = 2*u[a] +1
        z = 3*u[b] +1
        if y<z:
            u.append(y)
            a+=1
        elif y>z:
            u.append(z) 
            b+=1
        else:
            u.append(z)
            a+=1
            b+=1
    return u[n]